From 390493e5aab8a6c859a9d14020b6b9cd09bf6f57 Mon Sep 17 00:00:00 2001
From: Max Krummenacher <max.krummenacher@toradex.com>
Date: Fri, 30 Dec 2016 13:43:52 +0100
Subject: [PATCH] connman.service.in: don't start if nfs boot

connman reconnects its NIC's. If doing a NFS boot the rootfs
connection drops and the boot will fail.
As a simple workaround do not manage the network with connman
at all.
Alternatively one could black list the network interface used
for NFS in /etc/connman/main.conf.

Signed-off-by: Max Krummenacher <max.krummenacher@toradex.com>
---
 src/connman.service.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/connman.service.in b/src/connman.service.in
index dab48bc..72c0c91 100644
--- a/src/connman.service.in
+++ b/src/connman.service.in
@@ -7,6 +7,8 @@ After=dbus.service network-pre.target systemd-sysusers.service
 Before=network.target multi-user.target shutdown.target
 Wants=network.target
 Conflicts=systemd-resolved.service
+# only if not NFS mount, connman will disconnect your rootfs otherwise!
+ConditionKernelCommandLine=!root=/dev/nfs
 
 [Service]
 Type=dbus
-- 
2.12.0

